src/main/java/org/generationcp/middleware/api/germplasm/GermplasmServiceImpl.java

```diff
@@ -126,6 +132,305 @@ public GermplasmServiceImpl(final HibernateSessionProvider sessionProvider) {
+		final Germplasm germplasm, final Map<String, Object[]> conflictErrors, final GermplasmUpdateDTO germplasmUpdateDTO) {
+		for (final Map.Entry<String, String> codeValuesEntry : germplasmUpdateDTO.getNames().entrySet()) {
+			final String code = codeValuesEntry.getKey();
+			final String value = codeValuesEntry.getValue();
+			this.saveOrUpdateName(nameCodes, namesMap, germplasm, code, value,
+				conflictErrors);
+		}
+		for (final Map.Entry<String, String> codeValuesEntry : germplasmUpdateDTO.getAttributes().entrySet()) {
+			final String code = codeValuesEntry.getKey();
+			final String value = codeValuesEntry.getValue();
+			this.saveOrUpdateAttribute(attributeCodes, attributesMap, germplasm,
+				code, value, conflictErrors);
+		}
+
+	}
+
+	private void updatePreferredName(final Map<String, Integer> nameCodes, final Map<Integer, List<Name>> namesMap,
+		final Germplasm germplasm, final GermplasmUpdateDTO germplasmUpdateDTO, final Map<String, Object[]> conflictErrors) {
+		// Update preferred name
+		final Integer preferredNameTypeId = nameCodes.get(germplasmUpdateDTO.getPreferredName());
+		final List<Name> names = namesMap.get(germplasm.getGid());
+		final List<Name> preferredNames =
+			names.stream().filter(n -> n.getTypeId().equals(preferredNameTypeId)).collect(Collectors.toList());
+
+		if (preferredNames.size() == 1) {
+			for (final Name name : names) {
+				if (preferredNameTypeId != null) {
+					name.setNstat(name.getTypeId().equals(preferredNameTypeId) ? 1 : 0);
+					this.daoFactory.getNameDao().save(name);
+				}
+			}
+		} else if (preferredNames.size() > 1) {
+			conflictErrors.put("import.germplasm.update.preferred.name.duplicate.names", new Object[] {
+				germplasmUpdateDTO.getPreferredName(),
+				germplasm.getGid()});
+		} else if (!liquibase.util.StringUtils.isEmpty(germplasmUpdateDTO.getPreferredName())) {
+			conflictErrors.put("import.germplasm.update.preferred.name.doesnt.exist", new Object[] {
+				germplasmUpdateDTO.getPreferredName(),
+				germplasm.getGid()});
+		}
+	}
+
+	private void updateGermplasm(final Germplasm germplasm,
+		final Optional<Integer> locationIdOptional,
+		final Optional<Method> breedingMethodDtoOptional,
+		final Optional<Integer> germplasmDateOptional, final Optional<String> referenceOptional,
+		final Map<String, Object[]> conflictErrors) {
+
+		if (breedingMethodDtoOptional.isPresent()) {
+			final String oldMethodType = germplasm.getMethod().getMtype();
+			final String newMethodType = breedingMethodDtoOptional.get().getMtype();
+
+			// Only update the method if the new method has the same type as the old method.
+			if (this.isMethodTypeMatch(newMethodType, oldMethodType)) {
+				germplasm.setMethodId(breedingMethodDtoOptional.get().getMid());
+			} else {
+				conflictErrors.put("import.germplasm.update.breeding.method.mismatch", new Object[] {
+					germplasm.getGid(),
+					String.format("%s (%s)", germplasm.getMethod().getMname(), germplasm.getMethod().getMtype())});
```

Comment:
Is this format needed? I saw it when testing and it shows: 99,641 and the , is kind of weird. I think we can just convert the int to String without any format

Reply:
I can't replicate the issue you mentioned. Have you tried to recompile? The string format is needed to combine the method name and the type so that the message will show like this:

_Cannot change method for germplasm **1**, the breeding method should be the same type as **Unknown bulking method (DER**)_